### ISSUE TITLE
Create additional filters for selecting projects within CIP Planner

### ISSUE THREAD

**Issue by jim-z** [fpdcc/ccfp-asset-dashboard#257]
Need the ability to sort projects by phase year.  Currently can only sort by "year" but annual CIP is primarily based on the more specific phased funding years/amounts.
![image](https://github.com/fpdcc/ccfp-asset-dashboard/assets/11774880/84ee4bb0-3c48-4012-841f-7d528d9edcde)

**Comment by xmedr** [comment]
Short term: We'll be creating a data pool to help more easily create a 5-year plan by Thursday

Long term: We'll implement a way to sort phases by the year of their fundings, not just the year of the phase itself

**Comment by smcalilly** [comment]
@jim-z just to clarify, the year sort is based on the phase year, and y'all need to be able to sort based on the phase funding year?

in that case, would it be okay if we changed the UI for this? specifically, from your above screenshot, the relationship between the project's phase and the sub-table with the phase funding. in order to sort, i think we'll need to flatten the rows so that each phase funding source creates a unique row instead of containing that data in a sub-table.

the table would roughly look like this:
|phase|phase_year|funding_source|funding_amount|funding_year|project_description|
|------|------------|---------------|-----------------|-------------|--------------------|
|phase_1|2023|grants|$200k|2023|Construction following feasibility study from Engineering Contract|
|phase_1|2023|capital improvements|$70k|2024|Construction following feasibility study from Engineering Contract|

note the duplication of data in the columns `phase`, `phase_year`, `project_description`. this change would create duplicated data in the CIP tables, including the other phase and project fields that i omitted for brevity. the unique data in each row would be the phase funding, and all of the phase and project fields would be duplicated across each row. we could figure out a way to truncate the longer text fields so it's a better experience visually.

this change would be the best way to structure this data so we can filter on it in the table, and structuring the data like that would make the data better work with it after y'all export it.

i'm open to other ways of addressing this problem, but as it's currently designed it's likely we'll need to hack it in order to get the sorting to work on the phase funding fields.

**Comment by smcalilly** [comment]
@jim-z or, would a dedicated filter be a better solution? rather than using the table sort, we could create a filter similar to the "filter by section". if we did this then we might need to adjust the UI so those filters are placed in a way to that connects them to the table in a more obvious way.

**Comment by jim-z** [comment]
@smcalilly correct, my understanding is that the sort field is currently coming from here (see below) and that information is sort of irrelevant/redundant because the phase funding fields below that are more detailed and capture the information that actually gets added into the CIP:
![image](https://github.com/fpdcc/ccfp-asset-dashboard/assets/11774880/9fec8ea0-0b13-4042-9c98-ac96e17b1bdb)

Yes, I think duplicate rows with the same project description, etc. will need to be generated where projects include multiple phase funding entries.  The csv export should also work this way so that the information can be grabbed and transposed into our excel spreadsheet.  I think we should maybe eliminate the expanded dropdown table and use the phase fund year instead of phase year when displaying project information.

A dedicated filter at the top to select for phase funding year might be a good solution.  I think maybe removing the current year from what is displayed and having the dropdown subtable for phase funding automatically visible (rather than clicking the little drop down arrow) might make it a little more intuitive when searching for and selecting projects to add to portfolio:
![image](https://github.com/fpdcc/ccfp-asset-dashboard/assets/11774880/5220572a-3648-47cf-8d66-739cb16dfe21)

**Comment by smcalilly** [comment]
I've got the logic implemented for the filtering, we just need an okay for the UI change in #271

**Comment by jim-z** [comment]
You have my ok to implement the UI change to support filtering enhancements.  Just want to confirm, will Year now filter on phase year or how it was orignally set up with project year?  I invited our director to next week's meeting to discuss these changes and anything additional she might want to see in the UI, FYI.
